lib/validator/validator.go

```diff
@@ -519,4 +532,13 @@ func validatePolicySpec(v *validator.Validate, structLevel *validator.StructLeve
 		structLevel.ReportError(reflect.ValueOf(m.EgressRules),
 			"PolicySpec.EgressRules", "", reason("PreDNAT PolicySpec cannot have any EgressRules"))
 	}
+
```

Comment:
We probably also want the ability to validate the entire list of policy types beyond just the individual value validation done above.

i.e. I think this should be an error:

```
types:
- egress
- egress
```